backend/engine/plugins/ghas_secrets/main.py

```diff
@@ -59,45 +60,18 @@ def _ghas_secrets(gh: GitHubAPI, path: str) -> Tuple[list[dict], dict]:
     for alert in alerts:
         locations = _get_locations(gh, alert["number"])
         for location in locations:
-            valid, author, author_timestamp = _validate_location(location, path)
+            valid, author, author_timestamp = validate(location, path)
             if not valid:
                 LOG.debug("Location not valid for this branch")
                 continue
             item_id = str(uuid.uuid4())
-            item = {
-                "id": item_id,
-                "filename": location["details"]["path"],
-                "line": location["details"]["start_line"],
-                "commit": location["details"]["commit_sha"],
-                "type": _normalize_secret_type(alert["secret_type"]),
-                "author": author,
-                "author-timestamp": author_timestamp,
-                "validity": alert["validity"],
-                "state": alert["state"],
-                "created_at": alert["created_at"],
-            }
+            item = format_secret(gh, item_id, location, alert, author, author_timestamp)
             results.append(item)
             event_info[item_id] = {"match": alert["secret"], "type": item["type"]}
```

Comment:
`format_secret()` can now return an empty `dict`, so need to check for that before accessing `item["type"]` here (or use something like `item.get("item", some_default_value)`).